force-app/main/default/classes/TestDebuggingTest.cls

```diff
@@ -0,0 +1,629 @@
+        testLeads.add(new Lead(LastName = 'Shmo 7',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 7',  LeadSource = 'Web', Email = '', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 8',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 8',  LeadSource = '',    Email = 'Shmo8@gmail.com', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 9',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 9',  LeadSource = '',    Email = '', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 10', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 10', LeadSource = 'Web', Email = 'Shmo10@gmail.com', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 11', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 11', LeadSource = 'Web', Email = '', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 12', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 12', LeadSource = '',    Email = 'Shmo12@gmail.com', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 13', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 13', LeadSource = '',    Email = '', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 14', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 14', LeadSource = 'Web', Email = 'Shmo14@gmail.com', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 15', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 15', LeadSource = 'Web', Email = '', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 16', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 16', LeadSource = '',    Email = 'Shmo16@gmail.com', Phone = '', Industry = 'Technology'));
+        
+        insert testLeads;
+
+        // Verify 
+        List<Lead> updatedLeads = [SELECT LastName, LeadSource, Email, Phone, Industry, Lead_Score__c FROM Lead WHERE Id IN :testLeads];
+
+        for (Lead ld : updatedLeads) {
+            if ((ld.LeadSource != 'Web' && ld.Email == null && ld.Phone == null && ld.Industry != 'Technology') ||           // Score should be 0
+                (ld.LeadSource == 'Web' && ld.Email == null && ld.Phone == null && ld.Industry != 'Technology') ||
+                (ld.LeadSource != 'Web' && ld.Email != null && ld.Phone == null && ld.Industry != 'Technology')) {   
```

Comment:
This works but is pretty rigid in how it looks for the values. 
When I do something like this I will put the value I'm looking for in another field like Lead First Name and check if that's correct.